numpy/linalg/_linalg.py

```diff
@@ -152,8 +152,8 @@ def _commonType(*arrays):
     for a in arrays:
         type_ = a.dtype.type
         if issubclass(type_, inexact):
-            if isComplexType(type_):
-                is_complex = True
+            is_complex = is_complex or isComplexType(type_)
```

Comment:
Is this really worth doing? It takes my brain longer to process and shouldn't matter much performance-wise?

Reply:
I'm also not sure this routine is worth it, but if one goes for it, I'd start with something like
```
types = set(a.dtype.type for a in arrays)
```
which will generally reduce the number already, and then do
```
is_complex = any(isComplexType(type_) for type_ in types)
```
and something along similar lines for `result_type` (but with a check whether one can really not simply use the built-in `np.result_type` - not obvious to me).

But I'd do it in a separate PR.

Reply:
Interesting idea! It might not work out because the common use case is with ` len(arrays)`  just 1 or 2 and the set overhead is too large. I will remove this change here and test out in a new PR.